lab-nicole/route/mp3-router.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const AWS = require('aws-sdk');
+const multer = require('multer');
+const Router = require('express').Router;
+const debug = require('debug')('mp3uploader:mp3-router');
+const createError = require('http-errors');
+
+const Mp3 = require('../model/mp3files.js');
+const Playlist = require('../model/playlist.js');
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const mp3Router = module.exports = new Router();
+
+AWS.config.setPromisesDependency(require('bluebird'));
+
+const s3 = new AWS.S3();
+const dataDir = path.join(__dirname, '..', 'data');
+const upload = multer({ dest: dataDir });
+
+function s3uploadProm(params) {
```

Comment:
s3 has promises natively, and you can use them by chaining `.promise()` to the end of the s3 call